### DISCUSSION TITLE
Multiples Saves

### DISCUSSION THREAD

**Discussion by RPBarreto** [Vdauphin/HeartsAndMinds#1362] [Question and answer] [answered]
Hello,

I have a question, does H&M db supports having two missions(2 .pbos) in the same map with small differences and keep the save of both versions?
If yes, could you give me a insight on how to approach the issue and if I'm in the right direction, I'm fairly new to arma mission editing/making so the problem might be there :p

What I've tried:
Different pbo names;

Setting a name, on the empty string after the second comma in the files inside the db folder that have the params : ["_name", worldName, [""]]
Changing it to, for example: ["_name", worldName, ["test"]];

Changing All profileNamespace variable names in the db files, for example from btc_hm_%1_cities to btc_hmtest_%1_cities, this does seem to work but I haven't tested all the features of the mission to make sure it's working

Would that be the solution?

**Comment by Vdauphin** [comment, marked as answer]
Hello,

You nearly found how to do it

You need to update:
- https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/delete.sqf#L26 
- https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/save.sqf#L24 
- https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L24

And switch those lines from
```sqf 
["_name", worldName, [""]]
```
to this for exemple
```sqf 
["_name", "mission1", [""]]
```
with `"mission1"` for the first pbo and `"mission2"` for the second pbo

But I don't think it will work if you are running two servers with one the first pbo and the second with the second pbo at the same time. Because you could be in the case where both server could write at the same time in the same profile file :/
The data are saved in the server user profile file with extension : .vars.Arma3AlphaProfile (https://forums.bistudio.com/topic/143834-varsarma3alphaprofile/)
To avoid this you could just start your server with different profile name for each server [`-name=server1`](https://community.bistudio.com/wiki/Arma_3:_Startup_Parameters#Profile_Options). This also avoid the edition of files in `db` folder because now each pbo will save his data in a different profile

Cheers

**Reply by RPBarreto** [reply]
Thanks for the reply!

To clarify, do I keep the profileNamespace variables names changed?
Because I removed my changes, did the only example you showed me and I now have the issue of all vehicles under btc_vehicles dissapearing and intel zones appearing even without intel

edit: after looking abit more it isnt intel for caches, it's city liberation zones and that parameter is set as disabled and it still shows it.

I plan on running both missions on the same server and rotate between them, but if I end up using 2 I'll keep in mind to use different profiles.

**Reply by Vdauphin** [reply]
> Because I removed my changes, did the only example you showed

Yes just do what I did in my exemple

> I now have the issue of all vehicles under btc_vehicles dissapearing and intel zones appearing even without intel

It is because you make a mistake somewhere, check your .rpt for error and correct them until they disappear
Also delete the database to make sure you didn't already save a game with no vehicles inside
